config/routes.rb

```diff
@@ -1,4 +1,7 @@
 Rails.application.routes.draw do
   mount_devise_token_auth_for 'User', at: 'auth'
+
+  get 'books/:genre/:title/:author', to: 'books#show', as: :book
```

Comment:
Remember that you have the resource method that creates all the CRUD endpoints. Try using that instead.

http://guides.rubyonrails.org/routing.html#crud-verbs-and-actions

As we talked yesterday, the show endpoint should receive the id of the book